### ISSUE TITLE
Selenium grid does timesout

### ISSUE THREAD

**Issue by Watster69** [microsoft/playwright#16638]
Hi, running tests in Selenium Grid via docker simply doesn't work.

I've created the hub and node based on the documentation [HERE](https://playwright.dev/docs/selenium-grid#hub-and-nodes-mode)

Creating the hub....
`docker run -d -p 4442-4444:4442-4444 --name selenium-hub selenium/hub:latest`

From some googling the way to get the IP of the docker selenium-hub is to run this command
`docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' selenium-hub`

This IP address is then used in the docker commands as per the instructions.

Creating the node
```
docker run -d -p 5555:5555 --name chrome-1 \
    --shm-size="2g" \
    -e SE_EVENT_BUS_HOST=172.17.0.2 \
    -e SE_EVENT_BUS_PUBLISH_PORT=4442 \
    -e SE_EVENT_BUS_SUBSCRIBE_PORT=4443 \
    -e SE_NODE_GRID_URL="http://172.17.0.2:4444/wd/hub" \
    selenium/node-chrome:latest
```

I've set the environment variable as instructed on the documenation
`SELENIUM_REMOTE_URL=http://localhost:4444/wd/hub`

But when running it shows this error 

![image](https://user-images.githubusercontent.com/35819753/185391299-4ea676ef-e9c2-4baa-88b8-ea324582df64.png)

This being the line of code it's trying to run in the stacktrace...

```
globalThis.browser = await chromium.launch({
        // Not headless so we can watch test runs
        headless: false,
        // Slow so we can see things happening
        slowMo: 50,
});
```

After some googling it appears that the SELENIUM_REMOTE_URL should not contain `/wd/hub`. When I run it like this I get the following error

![image](https://user-images.githubusercontent.com/35819753/185391829-6bfd1451-805b-40f2-81be-6aa5e647f0a4.png)

I'm at a complete loss as to how to get this to work. Using the same selenium grid I can run tests successfully from my existing selenium project so I can't see anything wrong with the grid.

Does anyone have a working example of a grid setup that they have working?

**Comment by aslushnikov** [comment]
(over to Dima and assigning 1.26 so that we take a look on this later)

**Comment by wodyjowski** [comment]
Check if you can actually access `http://172.17.0.2:4444` in your browser. It looks to me like internal docker network address and from the stack trace I see that you're running tests locally (not in another container).
In this case a proper variable would be `-e SE_NODE_GRID_URL="http://localhost:4444`.
I would also recommend using `-e SE_EVENT_BUS_HOST=selenium-hub` to make use of docker's internall DNS tricks - the IP may change with every `up`.

**Comment by surecloud-rwattie** [comment]
@wodyjowski Thanks for your reply.

Yes I am running the tests locally and pointing to the selenium grid that is in the docker containers.

> In this case a proper variable would be -e SE_NODE_GRID_URL="http://localhost:4444.

I don't know how the internals of this all works but surely if I'm setting this on the node then whatever uses this grid url inside the node container will think it's localhost:4444 which in fact would be the node not the hub

> I would also recommend using -e SE_EVENT_BUS_HOST=selenium-hub to make use of docker's internall DNS tricks - the IP may change with every up.

I appreciate that, I had previously tried `selenium-hub` but was getting the errors so I thought I'd dig out the exact IP and it would have nothing to complain about.

Basically I've tried everything I can think of and still nothing works. I have a grid that is already in use for selenium tests and I'm trying to migrate to Playwright. I've migrated a couple of tests from our selenium project to the Playwright project and running against the same grid and still doesn't work so I know the grid is ok and can access our website etc. Something odd about the way playwright interacts with it.

If Selenium Grid is not suitable for Playwright then what is the best way to execute Playwright tests in parallel? We currently have 1100+ tests and want to execute them as quick as possible. Any help is greatly appreciated.

**Comment by Watster69** [comment]
Ok so I had tried it before I responded previously and still had no luck. However now using your content for the compose file, I now have it working! I'm honestly dumbfounded as to what's different other than I'm specifying ports for the hub as 4442-4444:4442-4444 and 5555:5555 for the nodes.

On a side note the documentation for this should be updated because the information [HERE](https://playwright.dev/docs/selenium-grid#hub-and-nodes-mode) clearly states `<selenium-hub-ip>` for the `SE_NODE_GRID_URL` which is what I was following.

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you disagree with this closure, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.
